painting/src/main/java/com/ataulm/artcollector/painting/ui/PaintingActivity.kt

```diff
@@ -27,12 +34,46 @@ class PaintingActivity : AppCompatActivity() {
         val paintingId = intent.data!!.pathSegments.last()
         injectDependencies(PaintingId(paintingId))
 
+        // TODO: check with app links that it still works and doesn't care about postponed transitions
+        postponeEnterTransition()
+
+        intent.loadImageIfAvailable()
         viewModel.painting.observe(this, DataObserver<Painting> { painting ->
             title = painting.title
             titleArtistTextView.text = "${painting.title} // ${painting.artist.name}" // TODO: string format, but also PaintingUiModel
             creditLineTextView.text = painting.creditLine ?: "Harvard Art Museum"
             creditLineTextView.setOnClickListener { startActivity(webIntent(painting.webUrl)) }
-            picasso.load(painting.imageUrl).into(imageView)
+            painting.loadImageIfDifferent()
```

Comment:
if the URL for the painting is different (maybe it's higher res in the painting API than in the gallery API), then we want to load this image.

I didn't handle the case where the other one takes ages to load and this one loads first - in that case, does Glide cancel the older request if the target is the same? I feel like it would :x: but should probably check.

Reply:
it does seem to 🤔 but reading the docs/code I can't see why, since it's creating a new Target for each load, and we're not keeping a reference to it and clearing it ourselves.

Reply:
cleared it explicitly anyway.

commit 57908982c42028c176548ed4f85934d4f80a4c21 (HEAD -> shared-element-transition, origin/shared-element-transition)
Author: Ataul Munim <hello@ataulm.com>
Date:   Sat Nov 24 13:55:44 2018 +0000

    Explicitly clear old requests in Glide before attempting the second load